File: frontend/src/lib/api/labels.ts
/** Impresión de etiquetas con código de barras (devuelve el PDF como Blob). */
import { fetchBlob } from "./client";

export interface LabelItem {
    product_id: string;
    copies: number;
}

export const labels = {
    pdf: (items: LabelItem[], showPrice: boolean) =>
        fetchBlob("/api/v1/inventory/labels/pdf", {
            method: "POST",
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify({ items, show_price: showPrice }),
        }),
};

File: backend/app/api/v1/schemas/labels.py
"""Schemas de impresión de etiquetas."""

from uuid import UUID

from pydantic import BaseModel


class LabelItem(BaseModel):
    product_id: UUID
    copies: int = 1


class LabelsRequest(BaseModel):
    items: list[LabelItem]
    show_price: bool = True
